### PR TITLE
Enhancement: Add message notification for data file loading process

### PR DESCRIPTION
**Pull request by YuZizhen**
Add message notification for data file loading process

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -415,6 +415,17 @@ _{Explain here how the data archiving feature will be implemented}_
 - Modify the pre-defined set of allowed characters to **capture any character instead**. This allows for flexibility since all special characters will be accepted for the `ADDRESS` field.
 - For addresses containing `/` characters, the user will be required to use the `\` character to escape each `/` character similar to the example in Planned Enhancement 1.
 
+### 3. Add message notification for data file loading process
```
Comment (KohGuanZeh) [inline docs/DeveloperGuide.md:418]: Do we still require this @Javiery3889 @Dethada @Arixeyeion if we are no longer planning to support editing of JSON file outside of FINDvisor?

Comment (YuZizhen) [reply docs/DeveloperGuide.md:418]: This one is for loading of data right? Like I save my contacts in the JSON then just copy over to a new computer?

Comment (Dethada) [reply docs/DeveloperGuide.md:418]: I think we should have it, useful in the case where the data file is corrupted for some reason, might not be due to manual modification.

Comment (Javiery3889) [reply docs/DeveloperGuide.md:418]: @KohGuanZeh @YuZizhen @Arixeyeion I agreww with what @Dethada mentioned, would be better if we include it

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -415,6 +415,17 @@ _{Explain here how the data archiving feature will be implemented}_
...
+**Current Implementation**: 
+
+- There is **no message notification** for incorrect data file loading and success data loading.
+- For example, when load a data file, the user does not know whether the data file is loaded succesfully into the FINDvisor unless the user check the contacts manually to see if there are any changes.
+
+**Proposed Enhancement**: 
+
+- Add a message for successful data file loading and an error message specifying the type of error encountered while loading data file. This provides a more direct feedback for the user.
+- For example, `Data file is loaded succesfully!` will be shown when the loading is succeeded. `Data file does not exist in the directory.` will be returned if the target datafile is not found in the directory.
```
[the remark this answers is not in the document]
Comment (YuZizhen) [reply docs/DeveloperGuide.md]: Ok! I have made the corresponding changes! Thanks :)

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -432,6 +432,17 @@ _{more aspects and alternatives to be added}_
...
 
+### 3. Notfiy user if data file is invalid
+**Current Implementation**: 
+
+- If the data file cannot be parsed due to invalid fields, FINDvisor will discard all data and start with an empty data file at the next run.
+- No notification will be displayed to the user when this occurs, which might affect the usability of FINDvisor.
+
+**Proposed Enhancement**: 
+
+- Notify the user if the data file is invalid on startup.
```
Comment (Javiery3889) [inline docs/DeveloperGuide.md:443]:
```suggestion
- Display a message in the Command Result Box if the data file cannot be parsed on startup.
```

**Review by Javiery3889 — CHANGES_REQUESTED** [review]

**Review by KohGuanZeh — COMMENTED** [review]
Just a nit on the phrasing. I think if the previous phrasing is preferred, the PR LGTM.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -432,7 +432,18 @@ _{more aspects and alternatives to be added}_
 - Modify the format of `DATE` to be `d-M-yyyy` and `DATETIME` to be `d-M-yyyy`T`H:mm` instead.
 - This allows FINDvisor to accept both single and double-digits day, month and hour values as valid `DATE` and `DATETIME` values and would not require users to pad these single digit values with a leading zero.
 
-### 3. Show warning to user when scheduling an overlapping meeting
+### 5. Notfiy user if data file is invalid
+**Current Implementation**: 
+
+- If the data file cannot be parsed due to invalid fields, FINDvisor will discard all data and start with an empty data file at the next run.
+- No notification will be displayed to the user when this occurs, which might affect the usability of FINDvisor.
```
[the remark this answers is not in the document]
Comment (YuZizhen) [reply docs/DeveloperGuide.md:439]: Yepp

**Review by KohGuanZeh — DISMISSED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -432,7 +432,18 @@ _{more aspects and alternatives to be added}_
...
+### 5. Notfiy user if data file is invalid
+**Current Implementation**: 
+
+- If the data file cannot be parsed due to invalid fields, FINDvisor will discard all data and start with an empty data file at the next run.
+- No error message is displayed to the user when this occurs and users are not informed of the data loss.
+
+**Proposed Enhancement**: 
+
+- Display a message in the Command Result Box if the data file cannot be parsed on startup.
+- For example, `Data file (<file_location>) could not be loaded!` will be shown when the data file is invalid, where `file_location` is the location of the current referenced data file.
```
Comment (Dethada) [inline docs/DeveloperGuide.md:444]: We should specify what we do with the invalid data file too, either still discard it or move it to somewhere.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -432,7 +432,20 @@ _{more aspects and alternatives to be added}_
...
+- Specific error message about what is the error will be shown at the command result box. The user can check the data file and modify it accordingly.
+- If new state changing command is made when the data file is invalid, the data file will be overwritten.
```
[the remark this answers is not in the document]
Comment (YuZizhen) [reply docs/DeveloperGuide.md:446]: Just a nit, should we mention modification of data file before overwritten of the data file?

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -430,7 +430,20 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+### 5. Notfiy user if data file is invalid
+**Current Implementation**: 
+
+- No error message is displayed to the user when this occurs and users are not informed of the data loss.
+- If new state changing command is made when the data file is invalid, the data file will be overwritten.
+
+**Proposed Enhancement**: 
+
+- Display a message in the Command Result Box if the data file cannot be parsed on startup.
+- For example, `Data file (<file_location>) could not be loaded!` will be shown when the data file is invalid, where `file_location` is the location of the current referenced data file.
```
Comment (Arixeyeion) [inline docs/DeveloperGuide.md:442]:
```suggestion
- For example, `Data file (<file_location>) could not be loaded!` will be shown in the command result box when the data file is invalid, where `file_location` is the location of the current referenced data file, so the user can modify it accordingly.
```

Then delete line 443.

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -430,7 +430,20 @@ The following activity diagram summarizes what happens when a user executes a ne
...
-### 5. Show warning to user when scheduling an overlapping meeting
+### 5. Notfiy user if data file is invalid
+**Current Implementation**: 
+
+- No error message is displayed to the user when this occurs and users are not informed of the data loss.
+- If new state changing command is made when the data file is invalid, the data file will be overwritten.
+
+**Proposed Enhancement**: 
+
+- Display a message in the Command Result Box if the data file cannot be parsed on startup.
```
Comment (KohGuanZeh) [inline docs/DeveloperGuide.md:441]:
```suggestion
- Display a message in the *Command Result Box* if the data file cannot be parsed on startup.
```
I think it is better to make it italics, given that it is how we will reference our UI components.

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (KohGuanZeh) [reply docs/DeveloperGuide.md:442]: I think this error message is sufficient but can be improved by changing `command result box` to `Command Result Box`. Line 443 should also be removed.

**Review by Arixeyeion — APPROVED** [review]
lgtm

**Review by KohGuanZeh — APPROVED** [review]

**Review by Javiery3889 — APPROVED** [review]

**Pull request merged**